app/services/foreman/plugin/role_lock.rb

```diff
@@ -9,10 +9,8 @@ def initialize(plugin_id)
 
       def register_role(name, permissions, role_registry, description = '')
         User.as_anonymous_admin do
-          Role.transaction do
-            role = process_role name, permissions, description
-            role_registry.role_ids << role.id
-          end
+          role = process_role name, permissions, description
+          role_registry.role_ids << role.id
```

Comment:
The reason removing the transaction fixes this is because the previous fix: https://github.com/theforeman/foreman/pull/6939/files is actually handling the race condition, but in a way that does not work within a transaction.  As soon as the error is hit, the transaction is aborted and any additional sql queries will fail.  

I can't see any reason this needs to be in a transaction personally since this all gets re-run at startup, but that may need some thought.